system/include/emscripten/promise.h

```diff
@@ -135,6 +135,20 @@ __attribute__((warn_unused_result)) em_promise_t emscripten_promise_any(
 __attribute__((warn_unused_result)) em_promise_t
 emscripten_promise_race(em_promise_t* promises, size_t num_promises);
 
+// Suspend Wasm execution until the given `promise` is resolved. Once the
+// promise is resolved, this function will return the value it was fulfilled
+// with. Since the stack is not unwound while Wasm execution is suspended, it is
+// safe to pass pointers to the stack to asynchronous work that is waited on
+// with this function. If the promise is rejected instead of fulfilled, the
+// suspended Wasm will never be resumed.
+//
+// When Wasm execution suspends, a JS promise will be returned from the Wasm
+// export that led to this call. That promise will be resolved once Wasm resumes
+// execution and returns normally from the export.
```

Comment:
I'm not sure you need this paragraph as that is basically describing what `ASYNCIFY` in general..  I don't think we want to do for each and every asyncify'd API do we?

Reply:
No, definitely not for every asyncified API, but maybe for this one because it exposes JSPI/Asyncify directly to the user without really doing anything else?

Reply:
Does it expose it directly to user though?   It seems rather like `dlopen` or `wget` in that its just a API that waits for something to be done before continuing?

Reply:
Yes, but for this API the thing it waits for is a promise being resolved, which is exactly how I would describe JSPI.